server/aesdsocket.c

```diff
@@ -11,61 +11,315 @@
+            t->writeout_buff = (char *)realloc(t->writeout_buff, sizeof(char) * t->writeout_buff_size);
+            if (t->writeout_buff == NULL) {
+                perror("realloc");
+                syslog(LOG_ERR, "realloc");
+                close(t->outfile_fd);
+                close(t->client_fd);
+                close(t->sock_fd);
+                free(t->writeout_buff);
+                free(t->readout_buff);
+                exit(EXIT_FAILURE);
+            }
+        }
+
+        if (sigprocmask(SIG_BLOCK, &sig_mask, NULL)) {
+            syslog(LOG_ERR, "sigprocmask");
+            close(t->outfile_fd);
+            close(t->client_fd);
+            close(t->sock_fd);
+            free(t->writeout_buff);
+            free(t->readout_buff);
             exit(EXIT_FAILURE);
         }
 
-        closelog();
-        exit(EXIT_SUCCESS);
+        if (full_packet_received) {
+            //Send back out full file
+
+            if (pthread_mutex_lock(t->mutex) != 0) {
+                perror("pthread_mutex_lock");
+                syslog(LOG_ERR, "pthread_mutex_lock");
+                close(t->outfile_fd);
+                close(t->client_fd);
+                close(t->sock_fd);
+                free(t->writeout_buff);
+                free(t->readout_buff);
+                exit(EXIT_FAILURE);
+            }
+
+            lseek(t->outfile_fd, 0, SEEK_SET);
+            
+            read_status = read(t->outfile_fd, &(t->readout_buff[readout_buff_index]), 1);
+            if (read_status == -1) {
+                perror("read");
+                syslog(LOG_ERR, "read");
+                close(t->outfile_fd);
+                close(t->client_fd);
+                close(t->sock_fd);
+                free(t->writeout_buff);
+                free(t->readout_buff);
+                exit(EXIT_FAILURE);
+            }
+
+
+            while (read_status != 0) {
+                if (t->readout_buff[readout_buff_index] == '\n') {
+                    send_status = send(t->client_fd, &(t->readout_buff[0]), readout_buff_index + 1, 0);
+                    if (send_status == -1) {
+                        perror("send");
+                        syslog(LOG_ERR, "send");
+                        close(t->outfile_fd);
+                        close(t->client_fd);
+                        close(t->sock_fd);
+                        free(t->writeout_buff);
+                        free(t->readout_buff);
+                        exit(EXIT_FAILURE);
+                    }
+
+                    readout_buff_index = 0;
+                }
+                else {
+                    readout_buff_index++;
+                }
+
+                if (readout_buff_index == t->readout_buff_size) {
+                    t->readout_buff_size *= 2;
+                    t->readout_buff = (char *)realloc(t->readout_buff, sizeof(char) * t->readout_buff_size);
+                    if (t->readout_buff == NULL) {
+                        perror("realloc");
```

Comment:
@vido2373 the error logging is good, but think about what someone other than you could do with these logs.  Something like "realloc failed allocating buffer of size %d in send/read thread loop" would be much more descriptive for anyone reading your code or your error logs.